### ISSUE TITLE
Clippy errors after crate in the workspace

### ISSUE THREAD

**Issue by pronebird** [rust-lang/rust-clippy#15075]
### Summary

When running `cargo clippy` it prints all warnings for all crates in the workspace.

However when running it with `-Dwarnings`, it errors after the first crate that contains error without showing all of the errors within the workspace.

```
cargo clippy --workspace --locked -- -Dwarnings
```

Instead it should collect all errors, print them, then error.

### Reproducer

I tried this code:

```rust
<code>
```

I expected to see this happen:

Instead, this happened:

### Version

```text
rustc 1.89.0-nightly (414482f6a 2025-05-13)
binary: rustc
commit-hash: 414482f6a0d4e7290f614300581a0b55442552a3
commit-date: 2025-05-13
host: aarch64-pc-windows-msvc
release: 1.89.0-nightly
LLVM version: 20.1.4
```

**Comment by blyxyas** [comment]
Doesn't this also happen with `cargo check`?
